### ISSUE TITLE
K.batch_dot work not right when dims >3

### ISSUE THREAD

**Issue by zwqjoy** [tensorflow/tensorflow#33709]
In TF 2.0 the  K.batch_dot  output dim is wrong
`import tensorflow.keras.backend as K
import tensorflow as tf

a = K.ones((7, 4, 2))
b = K.ones((7, 2, 5))
c = K.batch_dot(a, b)
print(c.shape)

a1 = K.ones((8, 7, 4, 2))
b1 = K.ones((8, 7, 2, 5))
c1 = K.batch_dot(a1, b1)
print("K.batch_dot(a1, b1): ", c1.shape)
c2 = tf.matmul(a1, b1)
print("tf.matmul(a1, b1): ", c2.shape)`

output:
`(7, 4, 5)
K.batch_dot(a1, b1):  (8, 7, 4, 7, 5)
tf.matmul(a1, b1):  (8, 7, 4, 5)
`

**Comment by gowthamkpr** [comment]
Was able to reploduce the issue. Heres my github [gist](https://colab.sandbox.google.com/gist/gowthamkpr/8b961afedcdb49de048431b503b7ae48/untitled211.ipynb).

**Comment by ymodak** [comment]
This is fixed with latest tf-nightly version '2.2.0-dev20200218'. Thanks!
```python
(7, 4, 5)
K.batch_dot(a1, b1):  (8, 7, 4, 5)
tf.matmul(a1, b1):  (8, 7, 4, 5)
```
